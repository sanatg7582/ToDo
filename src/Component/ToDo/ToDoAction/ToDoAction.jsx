import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrash, faPen } from "@fortawesome/free-solid-svg-icons";
import Button from "../../Shared/Button";
/**
 * This function return Edit and Delete button
 * @param {number} index
 * @param {func} handleOnClickRemove
 * @param {func} handleOnClickUpdate
 * @returns Button Element
 */
const ToDoAction = props => {
  const { index, handleOnClickRemove, handleOnClickUpdate } = props;
  return (
    <>
      <Button
        buttonType="button"
        btnClass="warning"
        handleOnClick={handleOnClickUpdate}
        index={index}
      >
        <FontAwesomeIcon icon={faPen} />
      </Button>
      &nbsp;
      <Button
        buttonType="button"
        btnClass="danger"
        handleOnClick={handleOnClickRemove}
        index={index}
      >
        <FontAwesomeIcon icon={faTrash} />
      </Button>
    </>
  );
};
// default props value
ToDoAction.defaultProps = {
  index: 0,
  handleOnClickRemove: () => {},
  handleOnClickUpdate: () => {}
};
//prop type validation
ToDoAction.propTypes = {
  index: PropTypes.number,
  handleOnClickRemove: PropTypes.func.isRequired,
  handleOnClickUpdate: PropTypes.func.isRequired
};
export default ToDoAction;
