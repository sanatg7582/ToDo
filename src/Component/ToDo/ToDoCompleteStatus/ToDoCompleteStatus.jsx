import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCheckCircle,
  faTimesCircle
} from "@fortawesome/free-solid-svg-icons";
import Button from "../../Shared/Button";

/**
 * This function return Is complete Button
 * @param {bool} isCompleteStatus,
 * @param {func} handleOnClick,
 * @param {number} index
 * @returns Element
 */
const ToDoCompleteStatus = props => {
  const { isCompleteStatus, handleOnClick, index } = props;
  return (
    <>
      {isCompleteStatus ? (
        <Button
          buttonType="button"
          btnClass="success"
          handleOnClick={handleOnClick}
          index={index}
          status={true}
        >
          <FontAwesomeIcon icon={faCheckCircle} />
        </Button>
      ) : (
        <Button
          buttonType="button"
          btnClass="danger"
          handleOnClick={handleOnClick}
          index={index}
          status={false}
        >
          <FontAwesomeIcon icon={faTimesCircle} />
        </Button>
      )}
    </>
  );
};
// default props value
ToDoCompleteStatus.defaultProps = {
  isCompleteStatus: false,
  handleOnClick: () => {},
  index: 0
};
// prop type validation
ToDoCompleteStatus.propTypes = {
  isCompleteStatus: PropTypes.bool.isRequired,
  handleOnClick: PropTypes.func.isRequired,
  index: PropTypes.number
};
export default ToDoCompleteStatus;
