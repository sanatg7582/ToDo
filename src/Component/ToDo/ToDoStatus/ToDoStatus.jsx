import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";
import Button from "../../Shared/Button";
/**
 * This function ToDo status button
 * @param {bool} todoStatus
 * @param {func} handleOnClick
 * @param {number} index
 * @return element
 */
const ToDoStatus = props => {
  const { todoStatus, handleOnClick, index } = props;
  return (
    <>
      {todoStatus ? (
        <Button
          buttonType="button"
          btnClass="success"
          handleOnClick={handleOnClick}
          index={index}
          status={true}
        >
          <FontAwesomeIcon icon={faCheck} />
        </Button>
      ) : (
        <Button
          buttonType="button"
          btnClass="danger"
          handleOnClick={handleOnClick}
          index={index}
          status={false}
        >
          <FontAwesomeIcon icon={faTimes} />
        </Button>
      )}
    </>
  );
};
// default props value
ToDoStatus.defaultProps = {
  todoStatus: true,
  handleOnClick: () => {},
  index: 0
};
// prop type validation
ToDoStatus.propTypes = {
  todoStatus: PropTypes.bool.isRequired,
  handleOnClick: PropTypes.func.isRequired,
  index: PropTypes.number
};
export default ToDoStatus;
