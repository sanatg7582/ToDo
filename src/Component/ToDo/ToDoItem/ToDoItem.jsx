import React from "react";
import PropTypes from "prop-types";
import Text from "../../Shared/Text";
import ToDOStatus from "../ToDoStatus";
import ToDoCompleteStatus from "../ToDoCompleteStatus";
import ToDoAction from "../ToDoAction";
/**
 * This function for todo Item row in table
 * @param {number} index
 * @param {string} todoHeading
 * @param {bool} todoStatus
 * @param {bool} todoIsCompleteStatus
 * @param {func} updateToDoStatus
 * @param {func} completeToDoStatus
 * @param {func} removeToDo
 * @param {func} updateToDo
 * @returns tr element
 */
const ToDoItem = props => {
  const {
    index,
    todoHeading,
    todoStatus,
    todoIsCompleteStatus,
    updateToDoStatus,
    completeToDoStatus,
    removeToDo,
    updateToDo
  } = props;
  return (
    <tr>
      <td>
        <Text strong={true} heading={todoHeading} />
      </td>
      <td>
        <ToDOStatus
          todoStatus={todoStatus}
          handleOnClick={updateToDoStatus}
          index={index}
        />
      </td>
      <td>
        <ToDoCompleteStatus
          isCompleteStatus={todoIsCompleteStatus}
          handleOnClick={completeToDoStatus}
          index={index}
        />
      </td>
      <td>
        <ToDoAction
          index={index}
          handleOnClickRemove={removeToDo}
          handleOnClickUpdate={updateToDo}
        />
      </td>
    </tr>
  );
};
// default props value
ToDoItem.defaultProps = {
  index: 0,
  todoHeading: "",
  todoStatus: true,
  todoIsCompleteStatus: false,
  updateToDoStatus: () => {},
  completeToDoStatus: () => {},
  removeToDo: () => {},
  updateToDo: () => {}
};
//props type validation
ToDoItem.propTypes = {
  index: PropTypes.number,
  todoHeading: PropTypes.string.isRequired,
  todoStatus: PropTypes.bool.isRequired,
  todoIsCompleteStatus: PropTypes.bool.isRequired,
  updateToDoStatus: PropTypes.func.isRequired,
  completeToDoStatus: PropTypes.func.isRequired,
  removeToDo: PropTypes.func.isRequired,
  updateToDo: PropTypes.func.isRequired
};
export default ToDoItem;
