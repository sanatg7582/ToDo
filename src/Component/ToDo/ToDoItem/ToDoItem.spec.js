import React from "react";
import { mount } from "enzyme";
import ToDoItem from "./ToDoItem";
import Text from "../../Shared/Text";
import ToDoStatus from "../ToDoStatus";
import ToDoCompleteStatus from "../ToDoCompleteStatus";
import ToDoAction from "../ToDoAction";
describe("Load <ToDoItem />", () => {
  let wrapper = "";
  const updateToDoStatus = jest.fn();
  const completeToDoStatus = jest.fn();
  const updateToDo = jest.fn();
  const removeToDo = jest.fn();
  beforeAll(() => {
    const props = {
      index: 0,
      todoHeading: "test",
      todoStatus: true,
      todoIsCompleteStatus: false,
      updateToDoStatus: updateToDoStatus,
      completeToDoStatus: completeToDoStatus,
      removeToDo: removeToDo,
      updateToDo: updateToDo,
      todo: {}
    };
    wrapper = mount(<ToDoItem {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });
  it("checking button length", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("checking first todo text and todo props ", () => {
    expect(
      wrapper
        .find(Text)
        .find("span")
        .text()
    ).toEqual("test");
    expect(
      wrapper
        .find(Text)
        .find("span")
        .hasClass("footerHeading")
    ).toEqual(true);
  });
  it("checking ToDoStatus props", () => {
    wrapper
      .find(ToDoStatus)
      .find("button")
      .simulate("click");
    expect(updateToDoStatus).toHaveBeenCalled();
    expect(
      wrapper
        .find(ToDoStatus)
        .find("button")
        .hasClass("btn success")
    ).toEqual(true);
    wrapper.setProps({ todoStatus: false });
    expect(
      wrapper
        .find(ToDoStatus)
        .find("button")
        .hasClass("btn danger")
    ).toEqual(true);
    expect(wrapper.find("tr")).toHaveLength(1);
  });

  it("checking ToDoCompleteStatus props", () => {
    wrapper
      .find(ToDoCompleteStatus)
      .find("button")
      .simulate("click");
    expect(completeToDoStatus).toHaveBeenCalled();
    expect(
      wrapper
        .find(ToDoCompleteStatus)
        .find("button")
        .hasClass("btn danger")
    ).toEqual(true);
    wrapper.setProps({ todoIsCompleteStatus: true });
    expect(
      wrapper
        .find(ToDoCompleteStatus)
        .find("button")
        .hasClass("btn success")
    ).toEqual(true);
    expect(wrapper.find("tr")).toHaveLength(1);
  });

  it("checking ToDoAction edit button", () => {
    expect(wrapper.find(ToDoAction).find("button")).toHaveLength(2);
    expect(wrapper.find("tr")).toHaveLength(1);
    wrapper
      .find(ToDoAction)
      .find("button")
      .first()
      .simulate("click");
    expect(updateToDo).toHaveBeenCalled();
  });
  it("checking ToDoAction remove button", () => {
    wrapper
      .find(ToDoAction)
      .find("button")
      .last()
      .simulate("click");
    expect(removeToDo).toHaveBeenCalled();
  });
  it("SnapShot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
