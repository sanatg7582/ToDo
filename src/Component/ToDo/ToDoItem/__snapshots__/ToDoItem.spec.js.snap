// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load <ToDoItem /> SnapShot test 1`] = `ReactWrapper {}`;
